### PR TITLE
refactor: migrated settings nav item to svelte5

### PR DESCRIPTION
**Pull request by gastoner**
### What does this PR do?
Migrates settingsNavItem to svelte5

### What issues does this PR fix or reference?
Required for https://github.com/containers/podman-desktop-extension-ai-lab/pull/2246

### How to test this PR?

- [x]  Tests are covering the bug fix or the new feature

### CODE REVIEW

File: packages/ui/src/lib/settingsNavItem/SettingsNavItem.svelte [hunk 1]
```diff
@@ -2,14 +2,27 @@
...
-export let icon: IconDefinition | undefined = undefined;
-export let onClick: () => void = () => {};
+interface Props {
+  title: string;
+  href: string;
+  section: boolean;
+  expanded: boolean;
+  child: boolean;
+  selected: boolean;
+  icon: IconDefinition | undefined;
```
Comment (benoitf) [inline packages/ui/src/lib/settingsNavItem/SettingsNavItem.svelte]:
```suggestion
  icon?: IconDefinition;
```

Comment (benoitf) [reply packages/ui/src/lib/settingsNavItem/SettingsNavItem.svelte]: nit

**Review by benoitf — APPROVED** [review]

**Review by axel7083 — APPROVED** [review]
LGTM

**Review by benoitf — CHANGES_REQUESTED** [review]
it's still thinking onClick is not defined ? even if it's initialized line 24 ?

(for the onClick?.())

**Review by benoitf — APPROVED** [review]
it's still thinking onClick is not defined ? even if it's initialized line 24 ?

(for the onClick?.())

**Comment by gastoner** [comment]
@benoitf I've changed it now from `onClick?.()` to `onClick()`
The problem was in binding, I've initialized it with `false` and it should be empty as it is now

**Review by benoitf — APPROVED** [review]
👍

**Review by dgolovin — APPROVED** [review]

**Pull request merged**